doc/project/webassembly.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: doc
+title: Emitting JavaScript modules
+---
+
+# Experimental WebAssembly backend
+
+Since Scala.js 1.17.0, there is an *experimental* WebAssembly backend (Wasm for short).
+Under some conditions, it is designed to be a *drop-in* replacement for the usual JavaScript backend.
+
+## Experimental status
+
+Being experimental means that:
+
+* The Wasm backend may be removed in a future *minor* version of Scala.js (or moved to a separate plugin).
+* Future versions of Scala.js may emit Wasm that requires *newer* versions of Wasm engines, dropping support for older engines.
+
+However, we do *not* expect the *quality* of the Wasm backend to be any lower than the JS backend.
+Feel free to report any issue you may experience with the same expectations as for the JS backend.
+
+## Requirements
+
+The Wasm backend emits code with the following requirements:
+
+* A JavaScript host (i.e., we do not currently generate standalone Wasm)
+* A Wasm engine with support for:
+  * Wasm 3.0
```

Comment:
```suggestion
  * Wasm 2.0
```

?

Reply:
Wasm 2.0 does not contain `return_call` yet, which we also use. Wasm 3.0 does contain GC, so we could omit it in theory, but in practice IMO it matters a lot more that we need GC, since that has less widespread support.

Reply:
Ah, that makes sense